dask_expr/io/parquet.py

```diff
@@ -62,6 +68,309 @@ def normalize_pa_schema(schema):
     return schema.to_string()
 
 
+class ToParquet(Expr):
+    _parameters = [
+        "frame",
+        "path",
+        "fs",
+        "fmd",
+        "engine",
+        "offset",
+        "partition_on",
+        "write_metadata_file",
+        "name_function",
+        "write_kwargs",
+    ]
+
+    @property
+    def _meta(self):
+        return None
+
+    def _divisions(self):
+        return (None, None)
+
+    def _simplify_down(self):
+        return ToParquetBarrier(
+            ToParquetData(
+                *self.operands,
+            ),
+            *self.operands[1:],
+        )
+
+
+class ToParquetData(Blockwise):
+    _parameters = ToParquet._parameters
+
+    @cached_property
+    def io_func(self):
+        return ToParquetFunctionWrapper(
+            self.engine,
+            self.path,
+            self.fs,
+            self.partition_on,
+            self.write_metadata_file,
+            self.offset,
+            self.name_function,
+            self.write_kwargs,
+        )
+
+    def _divisions(self):
+        return (None,) * (self.frame.npartitions + 1)
+
+    def _task(self, index: int):
+        return (self.io_func, (self.frame._name, index), (index,))
+
+
+class ToParquetBarrier(Expr):
+    _parameters = ToParquet._parameters
+
+    @property
+    def _meta(self):
+        return None
+
+    def _divisions(self):
+        return (None, None)
+
+    def _layer(self):
+        if self.write_metadata_file:
+            append = self.write_kwargs.get("append")
+            compression = self.write_kwargs.get("compression")
+            return {
+                (self._name, 0): (
+                    apply,
+                    self.engine.write_metadata,
+                    [
+                        self.frame.__dask_keys__(),
+                        self.fmd,
+                        self.fs,
+                        self.path,
+                    ],
+                    {"append": append, "compression": compression},
+                )
+            }
+        else:
+            return {(self._name, 0): (lambda x: None, self.frame.__dask_keys__())}
+
+
+def to_parquet(
```

Comment:
NOTE: This function is mostly copied from `dd.DataFrame.to_parquet`. I didn't find any benefit in moving any option validation of filesystem/overwriting logic into an `Expr` class.